src/components/cluster-deployments/transitionCallbacks.ts

```diff
@@ -263,53 +258,42 @@ export const getOnSaveHostsSelection =
       );
 
       // add RESERVED_AGENT_LABEL_KEY to the newly selected agents
-      const newAgentIds = values.selectedHostIds.filter(
-        (agentId) => !oldReservedAgents.find((agent) => agent.metadata.uid === agentId),
+      const addAgents = agents.filter(
+        (a) =>
+          hostIds.includes(a.metadata.uid) &&
+          !Object.hasOwnProperty.call(a.metadata.labels, RESERVED_AGENT_LABEL_KEY),
       );
-      const reservedResults = await Promise.all(
-        newAgentIds
-          .map((agentId) =>
-            matchingAgents.find((agent: CIM.AgentK8sResource) => agent.metadata.uid === agentId),
-          )
-          // why filter()? The user can change labels after making a host-selection and than continue. Take selection of just those "visible ones".
-          .filter(Boolean)
-          .map((agent) => {
-            const newLabels = { ...(agent.metadata.labels || {}) };
-            newLabels[RESERVED_AGENT_LABEL_KEY] = reservedAgentlabelValue;
-            return k8sPatch(agentModel, agent, [
-              {
-                op: agent.metadata.labels ? 'replace' : 'add',
-                path: '/metadata/labels',
-                value: newLabels,
-              },
-              {
-                op: agent.spec?.clusterDeploymentName ? 'replace' : 'add',
-                path: '/spec/clusterDeploymentName',
-                value: {
-                  name: clusterDeployment.metadata.name,
-                  namespace: clusterDeployment.metadata.namespace,
-                },
+      await Promise.all(
+        addAgents.map((agent) => {
+          const newLabels = { ...(agent.metadata.labels || {}) };
+          newLabels[RESERVED_AGENT_LABEL_KEY] = reservedAgentlabelValue;
+          return k8sPatch(agentModel, agent, [
+            {
+              op: agent.metadata.labels ? 'replace' : 'add',
+              path: '/metadata/labels',
+              value: newLabels,
+            },
+            {
+              op: agent.spec?.clusterDeploymentName ? 'replace' : 'add',
+              path: '/spec/clusterDeploymentName',
+              value: {
+                name: clusterDeployment.metadata.name,
+                namespace: clusterDeployment.metadata.namespace,
               },
-            ]);
-          }),
+            },
+          ]);
+        }),
       );
 
       // TODO(mlibra): check for errors in the releasedResults and reservedResults
-      console.log('--- releaseResults (TODO): ', releaseResults);
```

Comment:
How is this handled?

Reply:
you mean errors ? Its not handled. TODO still applies, i just removed the logging

Reply:
these remainers were tracked just via those `TODO`s so far, they will be forgotten when just removing

Reply:
but we have a TODO comment above it 

```
// TODO(mlibra): check for errors in the releasedResults and reservedResults
```

why do we need console.logs ? Its just unnecessary noise getting through the code

Reply:
but I can return the logging if you prefer :)